apps/cloudbroker/base/cloudapi__disks/methodclass/cloudapi_disks.py

```diff
@@ -382,3 +394,84 @@ def deleteDisks(self, diskIds, reason, permanently=False, **kwargs):
         for diskId in diskIds:
             self.delete(diskId, False, permanently, reason=reason)
         return True
+    
+    @authenticator.auth(groups=["level1", "level2", "level3"])
+    def addType(self, id, description, vpool=None, cacheratio=None, snapshotable=True, **kwargs):
+        if cacheratio <= 0 and vpool:
+            raise exceptions.BadRequest(
+                "Cache Ratio must be larger than Zero"
+            )
+        dtype = self.models.disktype.searchOne({'id': id})
+        if dtype:
+            raise exceptions.BadRequest(
+                "Disk type with id: {} already exits, Please choose other name".format(id)
+            )
```

Comment:
you need to make a check there is not type with same vpool (which is assume is the prefix?)